### PR TITLE
Migrate and extend ODIS tests from mobile to ContractKit

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Move some ODIS tests from mobile to ContractKit
- Add new tests to kit ODIS utils
- Rewrite combiner e2e test to use ContractKit
- Related cleanup of the e2e tests

### Tested

With the e2e tests but we should redeploy to staging to test the kit version upgrade.

### Related issues

- #4467 

### Backwards compatibility

Yes

### CODE REVIEW

**Review by i1skn — APPROVED** [review]
👍

**Review by alecps — APPROVED** [review]
lgtm

File: packages/phone-number-privacy/combiner/jest.config.js [hunk 1]
```diff
@@ -3,4 +3,5 @@ const { nodeFlakeTracking } = require('@celo/flake-tracker/src/jest/config.js')
 module.exports = {
   preset: 'ts-jest',
   ...nodeFlakeTracking,
+  setupFilesAfterEnv: ['<rootDir>/jest_setup.ts'],
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/jest.config.js]: I know it's not pretty, but to keep the flake tracker working this line should be
```suggestion
  setupFilesAfterEnv: ['<rootDir>/jest_setup.ts', ...nodeFlakeTracking.setupFilesAfterEnv],
```

**Review by alecps — APPROVED** [review]
Looks good to me. Just one suggested change

File: packages/phone-number-privacy/combiner/jest.config.js (hunk 1 above)
Comment (jmrossy) [reply packages/phone-number-privacy/combiner/jest.config.js]: Good catch, thanks!

**Review by codyborn — APPROVED** [review]
Thanks for all the refactoring! This looks great

**Pull request merged**